src/pybamm/solvers/base_solver.py

```diff
@@ -543,7 +556,7 @@ def modulo_event(symbol, expr, num_events):
                 )
 
         def modulo_t_discon(symbol, expr, num_events):
-            value = expr.evaluate(0, model.y0.full(), inputs=inputs)
+            value = expr.evaluate(0, model.y0_list[0].full(), inputs=inputs)
```

Comment:
is this anything to worry about? Presumably the input parameters can modify the times of the discontinuity, but we only consider the first set of input parameters.